### PR TITLE
Allow choosing an image via zenity when no arguments are provided

### PR DESCRIPTION
**Pull request by LeonMatthes**
Makes it much easier to open an image when you have many images to choose from

### CODE REVIEW

File: platforms/unix/config/bin.squeak.sh.in [hunk 1]
```diff
@@ -53,4 +53,15 @@ case $PLATFORMLIBDIR in
 /lib|/usr/lib)	SVMLLP=/lib:/usr/lib;;
 *)				SVMLLP="$PLATFORMLIBDIR:/lib:/usr$PLATFORMLIBDIR:/usr/lib"
 esac
-LD_LIBRARY_PATH="$PLUGINS:$SVMLLP:${LD_LIBRARY_PATH}" exec $GDB "$BIN/squeak" "$@"
+
+# If no image file is provided as a command-line argument, allow the user to 
+# select the image file via a file-chooser dialog
+if which zenity &>/dev/null && [ $# -eq 0 ]; then
```
Comment (fniephaus) [inline platforms/unix/config/bin.squeak.sh.in]: Could you please swap left and right? No need to check if `zenity` is installed if there are more args anyway.

File: platforms/unix/config/bin.squeak.sh.in [hunk 2]
```diff
@@ -53,4 +53,15 @@ case $PLATFORMLIBDIR in
...
+  IMAGE=$(zenity --title 'Select an image' --file-selection --filename "${RESOURCES}/" --file-filter '*.image' --file-filter '*')
```
Comment (fniephaus) [inline platforms/unix/config/bin.squeak.sh.in]: `IMAGE` is also used for all args (below), so please rename to something like `VM_ARGS`.
Also, `${RESOURCES}` is not defined, maybe remove `--filename "${RESOURCES}/"`?

File: platforms/unix/config/bin.squeak.sh.in [hunk 3]
```diff
@@ -53,4 +53,15 @@ case $PLATFORMLIBDIR in
...
+else
+  IMAGE=$@
+fi
+
+echo "Using image: ${IMAGE}"
```
Comment (fniephaus) [inline platforms/unix/config/bin.squeak.sh.in]: Please remove this print message.

File: platforms/unix/config/squeak.sh.in [hunk 4]
```diff
@@ -53,4 +53,15 @@ case $PLATFORMLIBDIR in
 /lib|/usr/lib)	SVMLLP=/lib:/usr/lib;;
 *)				SVMLLP="$PLATFORMLIBDIR:/lib:/usr$PLATFORMLIBDIR:/usr/lib"
 esac
-LD_LIBRARY_PATH="$PLUGINS:$SVMLLP:${LD_LIBRARY_PATH}" exec $GDB "$BIN/squeak" "$@"
```
Comment (fniephaus) [inline platforms/unix/config/squeak.sh.in]: See comments for other file...

**Review by fniephaus — CHANGES_REQUESTED** [review]

**Comment by eliotmiranda** [comment]
Hi Leon,

    alas I think this way will fail:

if [ $# -eq 0 && which zenity &>/dev/null ]; then
	  VM_ARGS=$(zenity --title 'Select an image' --file-selection --file-filter '*.image' --file-filter '*')
	else
	  VM_ARGS=$@
	fi
	
	LD_LIBRARY_PATH="$PLUGINS:$SVMLLP:${LD_LIBRARY_PATH}" exec $GDB "$BIN/squeak" "${VM_ARGS}"

That's illustrated by the following test programme:
-----------8<-------------
#!/bin/bash
ARGS=$@
echo; echo A
for a in "$ARGS"; do
	echo $a
done
ARGS="$@"
echo; echo B
for a in "$ARGS"; do
	echo $a
done
ARGS="$@"
echo; echo C
for a in $ARGS; do
	echo $a
done
ARGS=$@
echo; echo D
for a in $ARGS; do
	echo $a
done
-----------8<-------------

If one invokes this with ./testshellargs.sh 'arg one' 'arg two' then the output is

A
arg one arg two

B
arg one arg two

C
arg
one
arg
two

D
arg
one
arg
two

We have to find some way of writing this so that the output is
YES!!!
arg one
arg two

**Comment by eliotmiranda** [comment]
Hi Leon, the only way I know of doing this is to expand $@ within double quotes at the point of command dispatch.  Here's the illustration of the magic incantation:

ZENITY_ARGS=
echo 'YES!!!'
for a in ${ZENITY_ARGS:="$@"}; do
        echo $a
done

so the right form would be

ZENITY_ARGS=
if [ $# -eq 0 && which zenity &>/dev/null ]; then
ZENITY_ARGS=$(zenity --title 'Select an image' --file-selection --file-filter '.image' --file-filter '')
fi

LD_LIBRARY_PATH="$PLUGINS:$SVMLLP:${LD_LIBRARY_PATH}" exec $GDB "$BIN/squeak" ${ZENITY_ARGS:="$@"}

**Pull request closed without merging**